// @formant/realtime-sdk/dist/protos/api/signaling/v1/signaling_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('@grpc/grpc-js');
var protos_api_signaling_v1_signaling_pb = require('../../../../protos/api/signaling/v1/signaling_pb.js');
var google_protobuf_wrappers_pb = require('google-protobuf/google/protobuf/wrappers_pb.js');

function serialize_v1_signaling_api_CreatePeerRequest(arg) {
  if (!(arg instanceof protos_api_signaling_v1_signaling_pb.CreatePeerRequest)) {
    throw new Error('Expected argument of type v1.signaling.api.CreatePeerRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_v1_signaling_api_CreatePeerRequest(buffer_arg) {
  return protos_api_signaling_v1_signaling_pb.CreatePeerRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_v1_signaling_api_CreatePeerResponse(arg) {
  if (!(arg instanceof protos_api_signaling_v1_signaling_pb.CreatePeerResponse)) {
    throw new Error('Expected argument of type v1.signaling.api.CreatePeerResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_v1_signaling_api_CreatePeerResponse(buffer_arg) {
  return protos_api_signaling_v1_signaling_pb.CreatePeerResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_v1_signaling_api_DeletePeerRequest(arg) {
  if (!(arg instanceof protos_api_signaling_v1_signaling_pb.DeletePeerRequest)) {
    throw new Error('Expected argument of type v1.signaling.api.DeletePeerRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_v1_signaling_api_DeletePeerRequest(buffer_arg) {
  return protos_api_signaling_v1_signaling_pb.DeletePeerRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_v1_signaling_api_DeletePeerResponse(arg) {
  if (!(arg instanceof protos_api_signaling_v1_signaling_pb.DeletePeerResponse)) {
    throw new Error('Expected argument of type v1.signaling.api.DeletePeerResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_v1_signaling_api_DeletePeerResponse(buffer_arg) {
  return protos_api_signaling_v1_signaling_pb.DeletePeerResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_v1_signaling_api_GetIceServersRequest(arg) {
  if (!(arg instanceof protos_api_signaling_v1_signaling_pb.GetIceServersRequest)) {
    throw new Error('Expected argument of type v1.signaling.api.GetIceServersRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_v1_signaling_api_GetIceServersRequest(buffer_arg) {
  return protos_api_signaling_v1_signaling_pb.GetIceServersRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_v1_signaling_api_GetIceServersResponse(arg) {
  if (!(arg instanceof protos_api_signaling_v1_signaling_pb.GetIceServersResponse)) {
    throw new Error('Expected argument of type v1.signaling.api.GetIceServersResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_v1_signaling_api_GetIceServersResponse(buffer_arg) {
  return protos_api_signaling_v1_signaling_pb.GetIceServersResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_v1_signaling_api_GetPeersRequest(arg) {
  if (!(arg instanceof protos_api_signaling_v1_signaling_pb.GetPeersRequest)) {
    throw new Error('Expected argument of type v1.signaling.api.GetPeersRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_v1_signaling_api_GetPeersRequest(buffer_arg) {
  return protos_api_signaling_v1_signaling_pb.GetPeersRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_v1_signaling_api_GetPeersResponse(arg) {
  if (!(arg instanceof protos_api_signaling_v1_signaling_pb.GetPeersResponse)) {
    throw new Error('Expected argument of type v1.signaling.api.GetPeersResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_v1_signaling_api_GetPeersResponse(buffer_arg) {
  return protos_api_signaling_v1_signaling_pb.GetPeersResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_v1_signaling_api_ReceiveSignalStreamRequest(arg) {
  if (!(arg instanceof protos_api_signaling_v1_signaling_pb.ReceiveSignalStreamRequest)) {
    throw new Error('Expected argument of type v1.signaling.api.ReceiveSignalStreamRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_v1_signaling_api_ReceiveSignalStreamRequest(buffer_arg) {
  return protos_api_signaling_v1_signaling_pb.ReceiveSignalStreamRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_v1_signaling_api_ReceiveSignalStreamResponse(arg) {
  if (!(arg instanceof protos_api_signaling_v1_signaling_pb.ReceiveSignalStreamResponse)) {
    throw new Error('Expected argument of type v1.signaling.api.ReceiveSignalStreamResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_v1_signaling_api_ReceiveSignalStreamResponse(buffer_arg) {
  return protos_api_signaling_v1_signaling_pb.ReceiveSignalStreamResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_v1_signaling_api_RefreshPeerRequest(arg) {
  if (!(arg instanceof protos_api_signaling_v1_signaling_pb.RefreshPeerRequest)) {
    throw new Error('Expected argument of type v1.signaling.api.RefreshPeerRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_v1_signaling_api_RefreshPeerRequest(buffer_arg) {
  return protos_api_signaling_v1_signaling_pb.RefreshPeerRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_v1_signaling_api_RefreshPeerResponse(arg) {
  if (!(arg instanceof protos_api_signaling_v1_signaling_pb.RefreshPeerResponse)) {
    throw new Error('Expected argument of type v1.signaling.api.RefreshPeerResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_v1_signaling_api_RefreshPeerResponse(buffer_arg) {
  return protos_api_signaling_v1_signaling_pb.RefreshPeerResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_v1_signaling_api_SendSignalRequest(arg) {
  if (!(arg instanceof protos_api_signaling_v1_signaling_pb.SendSignalRequest)) {
    throw new Error('Expected argument of type v1.signaling.api.SendSignalRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_v1_signaling_api_SendSignalRequest(buffer_arg) {
  return protos_api_signaling_v1_signaling_pb.SendSignalRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_v1_signaling_api_SendSignalResponse(arg) {
  if (!(arg instanceof protos_api_signaling_v1_signaling_pb.SendSignalResponse)) {
    throw new Error('Expected argument of type v1.signaling.api.SendSignalResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_v1_signaling_api_SendSignalResponse(buffer_arg) {
  return protos_api_signaling_v1_signaling_pb.SendSignalResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


var SignalingService = exports.SignalingService = {
  createPeer: {
    path: '/v1.signaling.api.Signaling/CreatePeer',
    requestStream: false,
    responseStream: false,
    requestType: protos_api_signaling_v1_signaling_pb.CreatePeerRequest,
    responseType: protos_api_signaling_v1_signaling_pb.CreatePeerResponse,
    requestSerialize: serialize_v1_signaling_api_CreatePeerRequest,
    requestDeserialize: deserialize_v1_signaling_api_CreatePeerRequest,
    responseSerialize: serialize_v1_signaling_api_CreatePeerResponse,
    responseDeserialize: deserialize_v1_signaling_api_CreatePeerResponse,
  },
  deletePeer: {
    path: '/v1.signaling.api.Signaling/DeletePeer',
    requestStream: false,
    responseStream: false,
    requestType: protos_api_signaling_v1_signaling_pb.DeletePeerRequest,
    responseType: protos_api_signaling_v1_signaling_pb.DeletePeerResponse,
    requestSerialize: serialize_v1_signaling_api_DeletePeerRequest,
    requestDeserialize: deserialize_v1_signaling_api_DeletePeerRequest,
    responseSerialize: serialize_v1_signaling_api_DeletePeerResponse,
    responseDeserialize: deserialize_v1_signaling_api_DeletePeerResponse,
  },
  refreshPeer: {
    path: '/v1.signaling.api.Signaling/RefreshPeer',
    requestStream: false,
    responseStream: false,
    requestType: protos_api_signaling_v1_signaling_pb.RefreshPeerRequest,
    responseType: protos_api_signaling_v1_signaling_pb.RefreshPeerResponse,
    requestSerialize: serialize_v1_signaling_api_RefreshPeerRequest,
    requestDeserialize: deserialize_v1_signaling_api_RefreshPeerRequest,
    responseSerialize: serialize_v1_signaling_api_RefreshPeerResponse,
    responseDeserialize: deserialize_v1_signaling_api_RefreshPeerResponse,
  },
  getPeers: {
    path: '/v1.signaling.api.Signaling/GetPeers',
    requestStream: false,
    responseStream: false,
    requestType: protos_api_signaling_v1_signaling_pb.GetPeersRequest,
    responseType: protos_api_signaling_v1_signaling_pb.GetPeersResponse,
    requestSerialize: serialize_v1_signaling_api_GetPeersRequest,
    requestDeserialize: deserialize_v1_signaling_api_GetPeersRequest,
    responseSerialize: serialize_v1_signaling_api_GetPeersResponse,
    responseDeserialize: deserialize_v1_signaling_api_GetPeersResponse,
  },
  getIceServers: {
    path: '/v1.signaling.api.Signaling/GetIceServers',
    requestStream: false,
    responseStream: false,
    requestType: protos_api_signaling_v1_signaling_pb.GetIceServersRequest,
    responseType: protos_api_signaling_v1_signaling_pb.GetIceServersResponse,
    requestSerialize: serialize_v1_signaling_api_GetIceServersRequest,
    requestDeserialize: deserialize_v1_signaling_api_GetIceServersRequest,
    responseSerialize: serialize_v1_signaling_api_GetIceServersResponse,
    responseDeserialize: deserialize_v1_signaling_api_GetIceServersResponse,
  },
  sendSignal: {
    path: '/v1.signaling.api.Signaling/SendSignal',
    requestStream: false,
    responseStream: false,
    requestType: protos_api_signaling_v1_signaling_pb.SendSignalRequest,
    responseType: protos_api_signaling_v1_signaling_pb.SendSignalResponse,
    requestSerialize: serialize_v1_signaling_api_SendSignalRequest,
    requestDeserialize: deserialize_v1_signaling_api_SendSignalRequest,
    responseSerialize: serialize_v1_signaling_api_SendSignalResponse,
    responseDeserialize: deserialize_v1_signaling_api_SendSignalResponse,
  },
  receiveSignalStream: {
    path: '/v1.signaling.api.Signaling/ReceiveSignalStream',
    requestStream: false,
    responseStream: true,
    requestType: protos_api_signaling_v1_signaling_pb.ReceiveSignalStreamRequest,
    responseType: protos_api_signaling_v1_signaling_pb.ReceiveSignalStreamResponse,
    requestSerialize: serialize_v1_signaling_api_ReceiveSignalStreamRequest,
    requestDeserialize: deserialize_v1_signaling_api_ReceiveSignalStreamRequest,
    responseSerialize: serialize_v1_signaling_api_ReceiveSignalStreamResponse,
    responseDeserialize: deserialize_v1_signaling_api_ReceiveSignalStreamResponse,
  },
};

exports.SignalingClient = grpc.makeGenericClientConstructor(SignalingService);
